db/sg_replicate_conflict_resolver.go

```diff
@@ -137,6 +140,38 @@ func (c *ConflictResolver) Resolve(ctx context.Context, conflict Conflict) (winn
 	return winner, ConflictResolutionMerge, err
 }
 
+// ResolveForHLV is a wrapper for ConflictResolverFunc that evaluates whether conflict resolution resulted in localWins,
+// remoteWins, or merge, specifically for HLV-aware conflict resolution.
+func (c *ConflictResolver) ResolveForHLV(ctx context.Context, conflict Conflict) (winner Body, resolutionType ConflictResolutionType, err error) {
+
+	winner, err = c.crf(ctx, conflict)
+	if err != nil {
+		return winner, "", err
+	}
+
+	winningRev, ok := winner[BodyCV]
+	if !ok {
+		c.stats.ConflictResultMergeCount.Add(1)
+		return winner, ConflictResolutionMerge, nil
+	}
+
+	localRev, ok := conflict.LocalDocument[BodyCV]
+	if ok && localRev == winningRev {
+		c.stats.ConflictResultLocalCount.Add(1)
+		return winner, ConflictResolutionLocal, nil
+	}
+
+	remoteRev, ok := conflict.RemoteDocument[BodyCV]
+	if ok && remoteRev == winningRev {
+		c.stats.ConflictResultRemoteCount.Add(1)
+		return winner, ConflictResolutionRemote, nil
+	}
+
+	base.InfofCtx(ctx, base.KeyReplicate, "Conflict resolver returned non-empty revID (%s) not matching local (%s) or remote (%s), treating result as merge.", winningRev, localRev, remoteRev)
```

Comment:
Does this indicate user error in the custom conflict resolver, or can/should we prevent the user from writing `_cv` by the user?

It seems like it would be nice if we could prevent this caveat from the docs https://docs.couchbase.com/sync-gateway/current/sync/sync-inter-syncgateway-conflict-resolution.html#conflict-resolution-approaches

> When creating a new revision, do not provide a _rev property. A new revision ID will be generated. Use delete mergedDoc._rev to invalidate the property.

Reply:
I agree this is probably preferable , will make this change.